src/main/java/org/eclipse/golo/runtime/RegularMethodFinder.java

```diff
@@ -82,37 +86,39 @@ public boolean isOverloaded() {
   }
 
   private boolean isValidPrivateStructAccess(Method method) {
-    Object receiver = invocation.arguments[0];
+    Object receiver = invocation.arguments()[0];
     if (!(receiver instanceof GoloStruct)) {
       return false;
     }
     String receiverClassName = receiver.getClass().getName();
     String callerClassName = callerClass.getName();
-    return method.getName().equals(invocation.name)
-      && isPrivate(method.getModifiers())
-      && (receiverClassName.startsWith(callerClassName)
-          || callerClassName.equals(reverseStructAugmentation(receiverClassName)))
-      && TypeMatching.argumentsMatch(method, invocation.arguments);
+    return method.getName().equals(invocation.name())
+        && isPrivate(method.getModifiers())
+        && (receiverClassName.startsWith(callerClassName)
+        || callerClassName.equals(reverseStructAugmentation(receiverClassName)))
+        && TypeMatching.argumentsMatch(method, invocation.arguments());
   }
 
   private static String reverseStructAugmentation(String receiverClassName) {
     return receiverClassName.substring(0, receiverClassName.indexOf(".types"))
-      + "$" + receiverClassName.replace('.', '$');
+        + "$" + receiverClassName.replace('.', '$');
   }
 
 
   private Stream<Method> findInMethods() {
```

Comment:
We indeed have a code duplication (e.g. in `FunctionCallSupport` and `Predefined::fun`), but AFAIR, these particular methods are specific to a particular finder, so I don't think we can really extract common code. Type matching, checking and filtering are also candidates to possible duplication (even if I tried to remove as much as possible).
BTW, I started to extract the naming logic into a separate utility class, but didn't integrated it in my previous refactoring PR. Indeed, we are doing special names manipulations (like `MyModule.types.MyStruct`, augmentations replacing `.` with `$`, etc.) in several separated places: IR construction, bytecode generation, runtime lookup to name a few.